ruma-api/src/lib.rs

```diff
@@ -254,6 +255,59 @@ pub trait OutgoingRequest {
     ) -> Result<http::Request<Vec<u8>>, IntoHttpError>;
 }
 
+/// An extension to `OutgoingRequest` which provides Appservice specific methods
+pub trait OutgoingRequestAppserviceExt: OutgoingRequest + Sized {
```

Comment:
I'm a bit surprised this needs `Sized` (I'm guessing it needs it, or why did you add it?) but `OutgoingRequest` doesn't. Probably good to just make `Sized` a supertrait of `OutgoingRequest`.

Reply:
Yep,

```
error[E0277]: the size for values of type `Self` cannot be known at compilation time
   --> ruma-api/src/lib.rs:265:9
    |
265 |         self,
    |         ^^^^ doesn't have a size known at compile-time
    |
help: consider further restricting `Self`
    |
269 |     ) -> Result<http::Request<Vec<u8>>, IntoHttpError> where Self: Sized {
    |                                                        ^^^^^^^^^^^^^^^^^
help: function arguments must have a statically known size, borrowed types always have a known size
    |
265 |         &self,
    |         ^
```

> Probably good to just make Sized a supertrait of OutgoingRequest

Done